kinto_client/session.py

```diff
@@ -0,0 +1,95 @@
+import time
+
+import requests
+from six.moves.urllib.parse import urlparse
+
+from kinto_client import utils
+from kinto_client.exceptions import KintoException
+
+
+def create_session(server_url=None, auth=None, session=None, retry=0,
+                   retry_after=None):
+    """Returns a session from the passed arguments.
+
+    :param server_url:
+        The URL of the server to use, with the prefix.
+    :param auth:
+        A requests authentication policy object.
+    :param session:
+        An optional session object to use, rather than creating a new one.
+    """
+    # XXX Refactor the create_session to take place in the caller objects.
```

Comment:
One of the purposes of this function was to return the session, rather than doing the check in all the callers. Seeing this note, I wonder what's the best approach.